tools/xctrunnertool/run.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env python3
+
+import argparse
+import sys
+import shutil
+import os
+
+from lib.logger import Logger
+from lib.shell import shell, cp_r
+from lib.model import Configuration
+from lib.plist_util import PlistUtil
+from lib.lipo_util import LipoUtil
+from lib.dependencies import FRAMEWORK_DEPS, PRIVATE_FRAMEWORK_DEPS, DYLIB_DEPS
+
+
+class DefaultHelpParser(argparse.ArgumentParser):
+    """Argument parser error."""
+
+    def error(self, message):
+        sys.stderr.write(f"error: {message}\n")
+        self.print_help()
+        sys.exit(2)
+
+
+def main(argv) -> None:
+    "Script entrypoint."
+    parser = DefaultHelpParser()
+    parser.add_argument(
+        "--name",
+        required=True,
+        help="Name for the merged test bundle.",
+    )
+    parser.add_argument(
+        "--platform",
+        default="iPhoneOS.platform",
+        help="Runtime platform. Default: iPhoneOS.platform",
+    )
+    parser.add_argument(
+        "--output",
+        required=True,
+        help="Output path for merged test bundle.",
+    )
+    parser.add_argument(
+        "--xctest",
+        required=True,
+        action="append",
+        help="Path to xctest archive to bundle.",
+    )
+    parser.add_argument(
+        "--verbose",
+        required=False,
+        default=False,
+        type=bool,
+        help="Enable verbose logging to console.",
+    )
+    args = parser.parse_args()
+
+    # Generator configuration
+    xcode_path = shell("xcode-select -p").strip()
+    config = Configuration(
+        name=args.name,
+        xctests=args.xctest,
+        platform=args.platform,
+        xctrunner_path=args.output,
+        xcode_path=xcode_path,
+        verbose_logging=args.verbose,
+    )
+
+    # Shared logger
+    log = Logger(config).get(__name__)
+
+    # Log configuration
+    log.info("Bundle: %s", config.xctrunner.app)
+    log.info("Platform: %s", config.platform)
+    log.info("Xcode: %s", config.xcode.path)
+    xctest_names = ", ".join([os.path.basename(x) for x in config.xctests])
+    log.info("XCTests: %s", xctest_names)
+    log.info("Output: %s", config.xctrunner.path)
+
+    # Copy XCTRunner.app template
+    log.info("Copying XCTRunner.app template to %s", config.xctrunner.path)
+    shutil.rmtree(
+        config.xctrunner.path, ignore_errors=True
+    )  # Clean up any existing bundle
+    cp_r(config.xctrunner.template_path, config.xctrunner.path)
+
+    # XCTRunner is multi-arch. When launching XCTRunner on arm64e device, it
+    # will be launched as arm64e process by default. If the test bundle is arm64e
+    # bundle, the XCTRunner which hosts the test bundle will fail to be
+    # launched. So removing the arm64e arch from XCTRunner can resolve this
+    # case.
+    lipo = LipoUtil()
+    lipo.remove_arch(
+        bin=f"{config.xctrunner.path}/{config.xctrunner.name}", arch="arm64e"
+    )
```

Comment:
To match other stripping, we should only retain architectures that are used in the tests themselves. Similar to: https://github.com/bazelbuild/rules_apple/blob/c64f2aa03959f7443dcd148a5b2407b32ae89daa/apple/internal/stub_support.bzl#L62-L73

Reply:
Done!